api/v1/gitrepository_types.go

```diff
@@ -175,6 +180,13 @@ type GitRepositoryVerification struct {
 	SecretRef meta.LocalObjectReference `json:"secretRef"`
 }
 
+type GitRepositoryProxy struct {
+	// SecretRef specifies the Secret containing the proxy address and optionally
+	// the proxy username and password.
+	// +required
+	SecretRef meta.LocalObjectReference `json:"secretRef"`
+}
```

Comment:
I don't have any ideas in mind, but I wanted to keep the door open, since there's no real harm/issue in having a wrapper around the `SecretRef`.

Reply:
cc: @hiddeco

Reply:
Discussed this with @darkowlzz last week, and we are of opinion that #821 created a discrepancy in which now multiple Secret references can be defined for an object. While historically (e.g. HelmRepository) accepts a single Secret reference with fixed key names.

I think in terms of what it is being used for (e.g. proxy configuration settings, which may be the same within a namespace and shared across multiple objects), this may actually be desirable. But I do also think that our APIs should align in terms of how we expect people to use them, which improves UX by reducing cognitive load.

Given this, I think:

1. We should settle on a generic way to configure certain common configurations, like proxies.
2. Ensure alignment between all object kinds on how such configurations are defined.
3. Change any kind within the Source API (and if applicable, other parts of Flux as well) to make use of this same structure. Preferably in a backwards compatible manner.

Reply:
This touches (but not directly) into the area of https://github.com/fluxcd/flux2/pull/3366

Reply:
After an on-call discussion, it was decided that shall drop the historical practice of accepting a single monolithic Secret reference and prefer specific dedicated Secret references.
`HelmRepository` shall align with `OCIRepository`'s API, i.e. a new field `.spec.certSecretRef` shall be added which would specify the Secret containing the TLS certificate data and support for the `certFile`, `keyFile`, `caFile` keys in the Secret specified in `.spec.secretRef` would be deprecated and undocumented while its still at `apiVersion: v1beta2`. Support for these keys will be fully removed when `HelmRepository` graduates to v1.

Reply:
#1131